src/plugin_nfacct.c

```diff
@@ -27,11 +27,14 @@ static int nfacct_callback(const struct nlmsghdr *nlh, void *data) {
 
         info("nfacct.plugin: increasing nfacct_list to size %d", size);
 
-        nfacct_list = realloc(nfacct_list, sizeof(struct nfacct_list) + (sizeof(struct mynfacct) * size));
-        if(!nfacct_list) {
+        struct nfacct_list *tmp_list = realloc(nfacct_list, sizeof(struct nfacct_list) + (sizeof(struct mynfacct) * size));
+        if(!tmp_list) {
             error("nfacct.plugin: cannot allocate nfacct_list.");
+            free(nfacct_list);
+            nfacct_list = NULL;
             return MNL_CB_OK;
         }
+        nfacct_list = tmp_list;
```

Comment:
`Summary: Common realloc mistake: 'nfacct_list' nulled but not freed upon failure`